src/config/tools/methods.ts

```diff
@@ -6,10 +6,8 @@ import {
 export function selectNetwork (chainID:any, type?: any) {
   return new Promise(resolve => {
     const { ethereum } = window
-    const ethereumFN:any = {
-      request: '',
-      ...ethereum
-    }
+    const ethereumFN: any = ethereum;
+    ethereumFN.request = (ethereum as any).request ?? '';
```

Comment:
calling `ethereumFN.request` will throw an error if `ethereum` obj is undefined. You dont need to set the `request` method as it should be part of the `ethereum` object you are referencing
```suggestion
    const ethereumFN: any = ethereum
```

Reply:
Yeah, this is definitely better. I originally raised an issue (#10) about this, I'm not sure why the code is de-structuring `ethereumFN` from `window.ethereum` at all, but I tried to keep my PR as close to the original code as possible. But I should have made lines 9 & 10 like this to properly maintain consistency:

```
const ethereumFN: any = ethereum ?? {};
ethereumFN.request = (ethereum as any).request ?? '';
```

Reply:
I simplified the code in my PR slightly which should also fix the concern with `window.ethereum` being undefined.